litgpt/adapter.py

```diff
@@ -151,7 +151,8 @@ def scaled_dot_product_attention(
         return y + self.gating_factor * ay
 
     def reset_parameters(self) -> None:
-        torch.nn.init.zeros_(self.gating_factor)
+        if hasattr(self, "gating_factor"):
```

Comment:
Good catch.
Adapter is added only at `config.adapter_start_layer`, so not all layers have it.